pkg/client/core/service_domain_client.go

```diff
@@ -114,61 +112,6 @@ func (s *domainService) DeletePrimaryHostedZone(ctx context.Context, opts client
 	return nil
 }
 
-func (s *domainService) CreatePrimaryHostedZone(_ context.Context, opts client.CreatePrimaryHostedZoneOpts) (*client.HostedZone, error) {
```

Comment:
Since we don't have two anymore, maybe you could rename the one called `CreatePrimaryHostedZoneWithoutUserinput` to `CreatePrimaryHostedZone`?

Reply:
Agreed!